appinfo/routes.php

```diff
@@ -2,10 +2,22 @@
 
 return [
 	'routes' => [
+
+		// enable CORS for api calls (API version 1)
+		['name' => 'api1#preflighted_cors', 'url' => '/api/1/{path}',
+			'verb' => 'OPTIONS', 'requirements' => ['path' => '.+']],
+
 		['name' => 'page#index', 'url' => '/', 'verb' => 'GET'],
 
 		['name' => 'tableTemplate#list', 'url' => '/table/templates', 'verb' => 'GET'],
 
+		// API
+		['name' => 'api1#index', 'url' => '/api/1/tables', 'verb' => 'GET'],
+
+		['name' => 'api1#data', 'url' => '/api/1/table/{tableId}', 'verb' => 'GET', 'postfix' => 'tableOnly'],
+		['name' => 'api1#data', 'url' => '/api/1/table/{tableId}/limit/{limit}', 'verb' => 'GET', 'postfix' => 'tableAndLimit'],
+		['name' => 'api1#data', 'url' => '/api/1/table/{tableId}/limit/{limit}/offset/{offset}', 'verb' => 'GET', 'postfix' => 'tableAndLimitAndOffset'],
```

Comment:
GET parameters will automatically be used to populate route method params, so `/api/1/table/{tableId}` is technically enough. You can add limit and offset via query params then.

Reply:
> If I just use the first line, I can not curl with limit and offset...

If you just used the first line you could `curl /api/1/table/{tableId}?limit={limit}&offset={offset}`

Reply:
:)